src/Library/Library.js

```diff
@@ -14,27 +16,78 @@ import { createSessionFromDemo } from "../sessions.js";
 import EntryRow from "./EntryRow.js";
 
 import illustration from "./library.svg";
+import { gettext as _ } from "gettext";
 
 import { build } from "../../troll/src/builder.js";
 
 export default function Library({ application }) {
   const objects = build(resource);
-  const { window, picture_illustration, search_entry } = objects;
+  const {
+    window,
+    picture_illustration,
+    search_entry,
+    dropdown_category,
+    dropdown_language,
+    results_empty,
+  } = objects;
   window.application = application;
   picture_illustration.set_resource(illustration);
+  makeDropdownFlat(dropdown_category);
+  makeDropdownFlat(dropdown_language);
 
   if (__DEV__) {
     window.add_css_class("devel");
   }
 
   let last_triggered;
+  let current_category = 0;
+  let current_language = 0;
 
   window.connect("close-request", quitOnLastWindowClose);
 
   const demos = getDemos();
   const widgets_map = new Map();
   const category_map = new Map();
+  const language_model = new Gtk.StringList();
+  const category_model = new Gtk.StringList();
+  language_model.append("All Languages");
+  category_model.append("All Categories");
+  const language_check = ["All Languages"];
+  const category_check = ["All Categories"];
```

Comment:
Sorry, I missed these, which should also be localized.

After that, I think any UI tweaks can happen when we circle back with code search.